### ISSUE TITLE
Tomogram refinement producing grid-looking artifact

### ISSUE THREAD

**Issue by mpm896** [MLI-lab/DeepDeWedge#2]
Hello,

I've been enjoying trying out DeepDeWedge to denoise/correct some of our tomograms. However, on our data I noticed that the losses (both training and validation) were **very** high compared to the tutorial data, and that after refining the tomograms it produces a weird, grid-like artifact. I thought it may be due to something wrong with input dimensions, etc and reassembling of the refined subtomograms into the full tomogram, however when I extracted subtomograms and reassembled them from an unrefined tomogram (using ddw functions), everything looked fine. This made me curious if the high loss of the model is just resulting in poor refinement of subtomogram edges?

I've attached screenshots below - the first is a screenshot of the training loss curves from fitting to the entire tomogram, the second is the training loss curves from fitting to a sub-region of the tomogram that contains mostly the cell of interest and excludes empty regions, and the third is a screenshot of the refined tomogram. As can be seen, removing the empty areas from the tomogram improves the trends of the training loss curves (I could have ran it for much longer), however the losses are still very high and it didn't seem to improve the refinement very much.

**Training curve from the full tomogram**
![image](https://github.com/MLI-lab/DeepDeWedge/assets/117948951/d342063b-6605-4d71-b984-c0105b936e13)

**Training curve for the tomogram region with most empty ice exluded**
![image](https://github.com/MLI-lab/DeepDeWedge/assets/117948951/01071bd6-efb0-4758-8464-de911321f81e)

**Snapshot of the refined tomogram**
![image](https://github.com/MLI-lab/DeepDeWedge/assets/117948951/56f8e4e9-bd06-4c8a-8f81-f6f3ada01080)

I also noticed that the tutorial used bin 6 data, whereas this data is bin 4 (pixel size ~1.06 nm). I'm rerunning it now with bin 6 data (pixel size ~1.6 nm) and the loss looks better, but it is still hovering above 900 so I'm not sure how the results will look. Let me know if you have any advice about this issue!

Best,
Matt

**Comment by SimWdm** [comment]
Hello Matt,

Thanks for trying DeepDeWedge and for opening this issue!

**1. Regarding the high losses:** Probably the voxel values of your data are higher than those in the tutorial dataset, resulting in higher absolute loss values. This should not be a problem, and I have already successfully refined tomograms where the losses were even higher than what you observed. 
It is more important to have a more or less stable decrease in both the fitting and the validation loss. The fitting loss curves in your two your plots look healthy, as does the validation loss curve in the second plot. The high variance of the validation loss in the full tomogram setup (first plot) is probably due to higher variance among the subtomograms. I don't think this is a big problem, as the fitting loss still behaves well, but I would prefer the setup from the second plot where you had excluded the empty areas. The curves in the second plot look pretty much ideal.

**2. Regarding the grid artifacts:** I have seen this problem before with an older version of DeepDeWedge, where it was caused by a bug in the tomogram refinement step. My first step towards solving your problem would be to find out if you are using that old version. (Unfortunately, the current version of the code that fixes the bug is not compatible with the old version.)

Does your version of the code include the file `DeepDeWedge/src/refine_tomogram.py`? If so, could you please share the body of the `refine_tomogram` function with me?

Best,
Simon

**Comment by mpm896** [comment]
Hi Simon,

I had actually updated ddw right before opening this issue - I reran the fitting and refinement and the curves look good and the refinement looks great despite the high losses. Thanks for letting me know about the reason behind high losses. I also no longer see the grid artifact, so that's great! Also, really nice improvements in usability of the program!

Best,
Matt

**Comment by SimWdm** [comment]
Hi Matt,

that's great news! Glad to hear that the artifacts are gone. Thanks also for the positive feedback :) 
If you ever have any other issues or questions, please don't hesitate to open another issue or send me an e-mail (simonw.wiedemann@tum.de).

Best,
Simon
